### PR TITLE
AdSpirit adapter for AMP

### PR DESCRIPTION
**Pull request by janwinkler**
Please review AdSpirit (http://www.adspirit.de) adapter for AMP.

### CODE REVIEW

File: ads/adspirit.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+   const i = document.createElement('ins');
+   i.setAttribute('data-asm-params', data['asm-params']);
+   i.setAttribute('data-asm-host', data['asm-host']);
+   i.setAttribute('class', 'asm_async_creative');
+   i.style.cssText = 'display:inline-block;text-align:left;';
+   global.document.getElementById('c').appendChild(i);
+
+   const s = document.createElement('script');
+   s.src = 'https://'+data[asm-host]+'/adasync.js';
```
Comment (lannka) [inline ads/adspirit.js]: missing quote around asm-host ?

File: ads/adspirit.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {checkData} from '../src/3p';
+
+/**
+  * @param {!Window} global
+  * @param {!Object} data
+  */
+export function adsense(global, data) {
```
Comment (lannka) [inline ads/adspirit.js]: please rename :-)

File: ads/adspirit.js (hunk 1 above)
Comment (lannka) [inline ads/adspirit.js]: code style: add space before and after '+'

**Comment by lannka** [comment]
Please run `gulp lint`

File: ads/adspirit.js (hunk 1 above)
Comment (janwinkler) [inline ads/adspirit.js]: done :-)

File: ads/adspirit.js (hunk 2 above)
Comment (janwinkler) [inline ads/adspirit.js]: done :-)

**Comment by janwinkler** [comment]
Hi lannka, im not really sure how this gulp thing works, can you do it or advise me?

**Comment by lannka** [comment]
@janwinkler you should be able to do eslint in your CLI with `gulp lint`. 
This will pick up the eslint config for this project. Online lint is not recommended here.
The travis build is failing because of some lint issues. Please get it fixed.

[Gulp](http://gulpjs.com/) is the build system we use. Check [here](https://github.com/ampproject/amphtml/blob/master/DEVELOPING.md#build--test) for all gulp commands supported in this project.

**Comment by janwinkler** [comment]
done, gulp lint gives no errors anymore

**Comment by janwinkler** [comment]
gulp presubmit is also fixed

File: ads/adspirit.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+import {checkData} from '../3p/3p';
+
+/**
+  * @param {!Window} global
+  * @param {!Object} data
+  */
+export function adspirit(global, data) {
+  checkData(data, ['asm-params', 'asm-host']);
```
Comment (lannka) [inline ads/adspirit.js]: use camel case.

**Comment by lannka** [comment]
Please make it working locally first.

File: ads/adspirit.js (hunk 3 above)
Comment (janwinkler) [inline ads/adspirit.js]: camel case for what?

**Comment by janwinkler** [comment]
im not really sure how to make it work locally? can you advise please?

**Comment by lannka** [comment]
Check out [DEVELOPMENT.md](https://github.com/ampproject/amphtml/blob/master/DEVELOPING.md). 

You basically

1. start a local server
2. open the `ads.amp.max.html` page in browser to verify adspirit ads load.

You can comment out all the other ad networks in `ads.amp.html` to get a clean test.

**Comment by lannka** [comment]
@janwinkler are you still on it? Please let me know when it's ready for review. Thanks.

**Comment by janwinkler** [comment]
yes im on it but having trouble to build it on my windows machine (symlinks do not work)

**Comment by janwinkler** [comment]
@lannka I finally got it build and ad is shown as it should

File: ads/adspirit.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+<amp-ad width=300 height=250
+    type="adspirit"
+    data-asm-params="&amp;pid=4"
+    data-asm-host="help.adspirit.de">
+</amp-ad>
+```
+
+## Configuration
+
+For semantics of configuration, please see ad network documentation.
```
Comment (lannka) [inline ads/adspirit.md]: Could you provide a link here?

**Comment by lannka** [comment]
It works great! LGTM with one comment.

File: ads/adspirit.md (hunk 4 above)
Comment (janwinkler) [inline ads/adspirit.md]: link added

File: ads/adspirit.md [hunk 5]
```diff
@@ -28,7 +28,7 @@ limitations under the License.
 
 ## Configuration
 
-For semantics of configuration, please see ad network documentation.
+For semantics of configuration, please see ad network documentation at http://help.adspirit.de/help/.
```
Comment (lannka) [inline ads/adspirit.md]: Nit pick: please leverage the markup :-)
`[ad network documentation](http://help.adspirit.de/help/).`

**Comment by lannka** [comment]
Never mind. I'll help you to format the link. I need to merge this PR before an upcoming refactory.

**Comment by janwinkler** [comment]
@lannka When will we see these changes live?

**Comment by lannka** [comment]
@janwinkler We do weekly release. It's cut in version 1469825155149, which is in canary now (for opt in user only). Will be in prod next week (if no blocking issue found).
You can check the [release history](https://github.com/ampproject/amphtml/releases) for any updates.

**Pull request merged**